api/types.proto

```diff
@@ -30,12 +30,27 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
+message Set {
+	message Void {}
+	map<string, Void> val = 1;
```

Comment:
There isn't a struct{}{} or interface{} in protobuf see: https://github.com/google/protobuf/issues/227

I could use struct.proto: https://github.com/google/protobuf/blob/a779ae609c9ca6fa37ee4acec2d48eaa9c3afb91/src/google/protobuf/struct.proto

However it feels ill-suited for the job, would add one more dependency, require to import google_protobuf when creating an element in a set, add 300 more lines to the go compilation and would endup creating another layer of abstraction that we don't have control over :)